sklearn/utils/estimator_checks.py

```diff
@@ -353,10 +356,22 @@ def _is_32bit():
     return struct.calcsize('P') * 8 == 32
 
 
+def gram_matrix_if_pairwise(X, estimator, kernel=linear_kernel):
+    if is_pairwise(estimator):
+        return kernel(X, X)
+    return X
+
+
 def check_estimator_sparse_data(name, estimator_orig):
+
+    # Sparse precomputed kernels aren't supported
+    if getattr(estimator_orig, 'kernel', None) == 'precomputed':
```

Comment:
Shouldn't you use is_pairwise here?

Actually I think we should be testing that an appropriate error is raised in to case